### ISSUE TITLE
On cluster build not working for a golang function

### ISSUE THREAD

**Issue by Shashankft9** [knative-sandbox/kn-plugin-func#809]
I am following this doc: https://github.com/knative-sandbox/kn-plugin-func/blob/main/docs/guides/on-cluster-build.md and have compiled func cli from main branch. 

I created function in go language. 
When I am doing `func deploy`, it fails with:
```
Error: function pipeline run has failed, please inspect logs of Tekton PipelineRun "my-function-git-pipeline-run-d92n4"
```
Upon taking a look at `step-create`, it says:
```
Paketo Go Build Buildpack 0.6.0
  Executing build process
    Running 'go build -o /layers/paketo-buildpacks_go-build/targets/bin -buildmode pie -trimpath .'
      Failed after 94ms
        go build: no main packages to build
failed to execute 'go build': exit status 1
ERROR: failed to build: exit status 1

```

I wonder if its because go functions require some scaffolding with boson go buildpack and that might not be included in the `create` step of `buildpacks` task here: https://raw.githubusercontent.com/tektoncd/catalog/master/task/buildpacks/0.3/buildpacks.yaml

**Comment by zroubalik** [comment]
@Shashankft9 you are right, that's the problem.

@lance did this PR: https://github.com/boson-project/packs/pull/5

We just need to set the proper builder for Golang based func projects.

**Comment by Shashankft9** [comment]
sure, related question - something like the PR you linked has to be done for rust runtime as well I believe? Since it uses one paketo community buildpack which is not included in the the paketo buildpacks builder.

**Comment by Shashankft9** [comment]
> Do you think you can handle Rust as well? Thanks!

Sure! but we will need to host the rust builder somewhere, just like the go builder. I don't know how I can do that, any pointers? 
Note: the rust builder also deletes the source files, so the PR #830 will unblock that. 

> This is a good question, we need to tackle this? Maybe the global configuration could solve this? Users could overwrite default buildpacks there? WDYT?

I think this might not be an immediate issue, but might come up for users later who are already familiar with buildpacks and want to use their custom ones in on cluster builds, maybe the global configuration work will address some of it - I will open a separate issue for this

**Comment by lance** [comment]
@jrangelramos do you think we can get an e2e test for this? I think it should work when using `func deploy --builder=host`.

**Comment by jrangelramos** [comment]
Yea. I think so. I can manage that.
